third_party/cronet_impl/sample_executor.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef COMPONENTS_CRONET_NATIVE_SAMPLE_SAMPLE_EXECUTOR_H_
+#define COMPONENTS_CRONET_NATIVE_SAMPLE_SAMPLE_EXECUTOR_H_
+// Cronet sample is expected to be used outside of Chromium infrastructure,
+// and as such has to rely on STL directly instead of //base alternatives.
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include "../../src/wrapper.h"
+// Sample implementation of Cronet_Executor interface using static
+// methods to map C API into instance of C++ class.
+class SampleExecutor {
```

Comment:
Does Cronet not provide a default executor in it's own library?

Reply:
No, as far as I can figure out. This is the reason why I did https://github.com/google/cronet.dart/pull/2#discussion_r659714798

I felt much more convenience handling the whole executor part in C/C++ rather than again creating a C function wrapper and passing it to dart. What do you recommend? keeping executor init in C/C++ or, exposing the method to dart?

Reply:
You could make a C API for interacting with the sample executor.

* a C function for creating one (returning some pointer to it)
* a C function for init-ing it (though, that might be combined with creating it)
* a C function for getting the `Cronet_ExecutorPtr` out,
* and then directly calling `Cronet_UrlRequest_InitWithParams` with the Cronet bindings rather than the wrapper.

That way you do not have to register the Cronet Cronet_UrlRequest_InitWithParams address in the wrapper.

Or am I missing something here? 😺

Reply:
Done :)